beater/beater.go

```diff
@@ -593,7 +594,7 @@ func (s *serverRunner) run(listener net.Listener) error {
 }
 
 // waitReady waits until the server is ready to index events.
-func (s *serverRunner) waitReady(ctx context.Context, kibanaClient kibana.Client) error {
+func (s *serverRunner) waitReady(ctx context.Context, kibanaClient kibana.Client, indexDocCountField *modelprocessor.IndexDocCountField) error {
```

Comment:
Can we separate concerns a little bit here? I don't think "waitReady" should know about model processors. Instead, can we have waitReady return the cluster info?